changelog/10606.bugfix.md

```diff
@@ -0,0 +1 @@
+Fix `max_history` truncation in `AugmentedMemoizationPolicy` to preserve the most recent `UserUttered` event.
```

Comment:
I'd also explain what the effect of this bug was, e.g.
```suggestion
Fix `max_history` truncation in `AugmentedMemoizationPolicy` to preserve the most recent `UserUttered` event.
Previously, `AugmentedMemoizationPolicy` failed to predict next action after long sequences of actions (longer than `max_history`) because the policy did not have access to the most recent user message.
```